src/main/java/Duke.java

```diff
@@ -1,10 +1,166 @@
+        Vector<Task> tasks = new Vector<>();
+
+        greeting();
+        printlnWithIndent(LONG_LINE);
+
+        Scanner in = new Scanner(System.in);
+        Boolean exit = false;
+
+        while (!exit) {
+            String line = in.nextLine();
+            String[] arguments = line.split(" ");
+
+            printlnWithIndent(LONG_LINE);
+
+            switch(arguments[0]) {
+            case "bye":
+                bye();
+                exit = true;
+                break;
+            case "list":
+                list(tasks);
+                break;
+            case "done":
+                done(tasks, arguments);
+                break;
+            case "deadline":
+                deadline(tasks, arguments);
+                break;
+            case "event":
+                event(tasks, arguments);
+                break;
+            case "todo":
+                todo(tasks, arguments);
+                break;
+            default:
+                save(tasks, line);
+            }
+
+            printlnWithIndent(LONG_LINE);
+        }
+        in.close();
+    }
+
+    // Print a greeting message when the program is invoked
+    protected static void greeting() {
+        printlnWithIndent("Hello! I'm Duke.");
+        printlnWithIndent("What can I do for you?");
+    }
+
+    // Print a goodbye message before the program exits
+    protected static void bye() {
+        printlnWithIndent("Bye. Hope to see you again soon!");
+    }
+
+    // Print out everything in the list, index starts from 1
+    protected static void list(Vector<Task> tasks) {
+        printlnWithIndent("Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i += 1) {
+            printlnWithIndent(String.format("%d.\t%s", i + 1, tasks.get(i)));
+        }
+    }
+
+    // Mark a task to be done with index specified in arguments[1]
+    protected static void done(Vector<Task> tasks, String[] arguments) {
+        if (arguments.length < 2) {
+            // An index must be provided for the task to be marked "done"
+            printlnWithIndent("You will need to give me an index, like this: `done 2`.");
+        } else {
+            try {
+                int index = Integer.parseInt(arguments[1]);
+                if (index > tasks.size() || index < 1) {
+                    // This index is out of the boundary of our database
+                    throw new IllegalArgumentException();
+                }
+
+                Task task = tasks.get(index - 1);
+                task.markAsDone();
+                tasks.set(index - 1, task);
+
+                printlnWithIndent("Nice! I've marked this task as done:");
+                printlnWithIndent("\t" + task);
+            } catch (NumberFormatException e) {
+                printlnWithIndent("Index provided is not a proper number.");
+            } catch (IllegalArgumentException e) {
+                printlnWithIndent("Task with this index is not found in our database.");
```

Comment:
is this an if-else-catch block? not sure if there is supposed to be a catch at the end of the if-else block

Reply:
This two catches  should not have a relation with the if-else. Exception thrown in the try block (no matter whether they are in if block or not) will be caught by these two catches.